src/main/java/Duke.java

```diff
@@ -1,10 +1,212 @@
+            System.out.println(Global.PATTERNLINE);
+
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(Global.PATTERNLINE + "\nFriend, You do not have so much task in the list\n"
+                    + Global.PATTERNLINE);
+        }
+
+    }
+
+    private static void addTodo(ArrayList<Task> tasks, String input, int count) throws DukeException{
+        if (input.length() < 6) {
+            throw new DukeException("☹ OOPS!!! The description of a Todo cannot be empty.\n" +
+                    "Please re-input or enter bye to terminate the program\n");
+        }
+        tasks.add(new ToDos(input.substring(5)));
+        replyLine(tasks, input, count);
+    }
+    private static void addDeadlines(ArrayList<Task> tasks, String input, int count) throws DukeException{
+        if(!input.contains(" ")) {
+            throw new DukeException("☹ OOPS!!! The description of a Deadlines cannot be empty.\n " +
+                    "Please re-input or enter bye to terminate the program\n");
+        } else if(input.length()-1 == input.indexOf(" ")) {
+            throw new DukeException("☹ OOPS!!! The description of a Deadlines cannot be empty.\n " +
+                    "Please re-input or enter bye to terminate the program\n");
+        }
+
+        int index = input.indexOf("/by");
+        if(index == -1) {
+            throw new DukeException("You have input your Schedule or Wrong format\nPlease include -> /by when\n");
+        }
+        String schedule = input.substring(index + 4);
+        String taskName = input.substring(input.indexOf(" ")+1, index);
+        tasks.add(new Deadlines(taskName, schedule));
+        replyLine(tasks, input, count);
+    }
+    private static void addEvents(ArrayList<Task> tasks, String input, int count) throws DukeException{
+        if(!input.contains(" ")) {
+            throw new DukeException("☹ OOPS!!! The description of a Events cannot be empty.\n " +
+                    "Please re-input or enter bye to terminate the program\n");
+        } else if(input.length()-1 == input.indexOf(" ")) {
+            throw new DukeException("☹ OOPS!!! The description of a Events cannot be empty.\n " +
+                    "Please re-input or enter bye to terminate the program\n");
+        }
+
+        int index = input.indexOf("/at");
+        String schedule, taskName;
+
+        if(index == -1) {
+           // System.out.println("You have not input your Schedule or Wrong format. \nPlease include -> /at when");
+            throw new DukeException("You have not input your Schedule or Wrong format. \nPlease include -> /at when\n");
+        }
+        schedule = input.substring(index + 4);
+        taskName = input.substring(input.indexOf(" ")+1, index);
+        tasks.add(new Events(taskName, schedule));
+        replyLine(tasks, input, count);
+    }
+
+    static Action validateCommand(String command) throws DukeException {
+        if(command.startsWith("list")) {
```

Comment:
The way of validating the user input commands are smart. And this method is able to be extended.